pkg/apis/workspace/v1alpha1/workspaceexposure_types.go

```diff
@@ -55,8 +55,13 @@ const (
 // +k8s:openapi-gen=true
 type WorkspaceExposureStatus struct {
 	// Workspace Exposure status
-	Phase            WorkspaceExposurePhase       `json:"phase,omitempty"`
-	ExposedEndpoints map[string][]ExposedEndpoint `json:"exposedEndpoints,omitempty"`
+	Phase            WorkspaceExposurePhase         `json:"phase,omitempty"`
+	ExposedEndpoints map[string]ExposedEndpointList `json:"exposedEndpoints,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+type ExposedEndpointList struct {
+	Items []ExposedEndpoint `json:"items,omitempty"`
```

Comment:
This seems a strange change to me, as the serialized map would look something like
```yaml
  status:
    exposedEndpoints:
      che-machine-exec:
        items:
        - name: che-machine-exec
          url: ws://serverb3e761b6331b4209-che-machine-exec-4444-devworkspace-controller.192.168.39.8.nip.io
          attributes: #...
      mysql: {}
      theia-ide:
        items:
        - name: theia
          url: http://serverb3e761b6331b4209-theia-ide-3100-devworkspace-controller.192.168.39.8.nip.io
          attributes: #...
```
with the extra nesting for `items`

Reply:
Agree. It was done because code-generator reports `map values must be named type`.
It's kind of quick build fix that I need to take a closer look at how we can get rid of it.
BTW it's non-official yet and we can review this part later. Maybe it would be better to rework it in the following way:
```
  exposedContainers:
    - name: che-machine-exec
      endpoints:
        - name: che-machine-exec
          url: ....
          attributes: #....
```

Reply:
Agree. It was done because code-generator reports `map values must be named type`.
It's kind of quick build fix that I need to take a closer look at how we can get rid of it.
BTW it's non-official yet and we can review this part later. Maybe it would be better to rework it in the following way:
```
  exposedContainers:
    - name: che-machine-exec
      endpoints:
        - name: che-machine-exec
          url: ....
          attributes: #....
```

Reply:
Now it's replaced with `type ExposedEndpointList []ExposedEndpoint`